devlib/utils/android.py

```diff
@@ -335,6 +335,14 @@ def execute(self, command, timeout=None, check_exit_code=False,
         try:
             return adb_shell(self.device, command, timeout, check_exit_code,
                              as_root, adb_server=self.adb_server, su_cmd=self.su_cmd)
+        except subprocess.CalledProcessError as e:
+            cls = TargetTransientCalledProcessError if will_succeed else TargetStableCalledProcessError
+            raise cls(
+                e.returncode,
+                command,
+                e.output,
+                e.stderr,
+            )
```

Comment:
I haven't spotted what the problem is yet, but testing with an ADB connection I get the following error:
```
2021-09-28 10:17:40,224 ERROR        signal:             self.target.grant_package_permission(self.apk_info.package, permission)
2021-09-28 10:17:40,224 ERROR        signal:           File "/mnt/1TBData/marc/Work/devlib/devlib/target.py", line 1717, in grant_package_permission
2021-09-28 10:17:40,224 ERROR        signal:             if 'is not a changeable permission type' in e.message:
2021-09-28 10:17:40,224 ERROR        signal:           File "/mnt/1TBData/marc/Work/devlib/devlib/exception.py", line 27, in message
2021-09-28 10:17:40,224 ERROR        signal:             if self._message is not None:
2021-09-28 10:17:40,224 ERROR        signal:
2021-09-28 10:17:40,226 DEBUG        signal:         Sending error-logged from <ErrorSignalHandler (DEBUG)>
2021-09-28 10:17:40,226 ERROR        signal:         AttributeError('TargetStableCalledProcessError' object has no attribute '_message')
```

Reply:
TargetCalledProcessError inherits its `__init__` from subprocess.CalledProcessError, which means `DevlibError.__init__` is not called. I fixed the issue by dealing with missing `_message` to keep it as close as possible to CalledProcessError.